src/components/WorkspaceDetails/WorkspaceDetails.test.tsx

```diff
@@ -64,6 +70,36 @@ describe('Workspace Details Page', () => {
       queryOptions: { path: 'demo-workspace' },
     });
   });
+  describe('Action menu', () => {
+    it('delete workspace', async () => {
+      k8sGetResourceMock.mockResolvedValue(workspaceData);
+      k8sDeleteResourceMock.mockResolvedValue({});
+      render(
+        <Router location={history.location} navigator={history}>
+          <WorkspaceDetails />
+        </Router>,
+      );
+      await waitFor(() => expect(k8sGetResourceMock).toHaveBeenCalledTimes(1));
+
+      // Open delete modal
+      fireEvent.click(screen.getByText('Actions'));
+      expect(await screen.findByText('Delete workspace')).toBeInTheDocument();
+      fireEvent.click(screen.getByText('Delete workspace'));
+
+      // Delete workspace via the modal
+      expect(await screen.findByRole('dialog')).toBeInTheDocument();
+      expect(await screen.findByRole('textbox')).toBeInTheDocument();
+      fireEvent.change(screen.getByRole('textbox'), {
+        target: { value: 'demo-workspace' },
+      });
+      fireEvent.click(screen.getByText('Delete'));
```

Comment:
Nit: not blocking - maybe we could eventually use `data-` tags in HTML to ensure we get back the elements we intend

Reply:
Bryan and I talked through this one quite a bit.  Just adding additional information from the conversation.  React Testing Library states that you should do what users are going to do.  This means that text matching is preferred over any attribute matching.  It also prevents adding test-only logic/structure into your code.

With that said, it can make your tests more brittle because if there are two more elements with the same text, this test will break.  (getByText will fail if there is not exactly 1 match). 

 In this case, so far it isn't a big deal so we aren't requiring a change for this PR, but in the future, the use of `.getByTestId` is preferred over `.getByText`.   See https://testing-library.com/docs/queries/bytestid/ for implementation information.